### DISCUSSION TITLE
Add WIFI driver support for  Cypress CYW43XXX

### DISCUSSION THREAD

**Discussion by mariopaja** [zephyrproject-rtos/zephyr#62136] [General]
Add WIFI driver support for Cypress CYW43XXX (spi/sdio). It seems like several boards (rpi pico w, arduino giga r1 wifi, infineon PSoC6 ...) are using this wifi module. Additionally, it seems like this module is popular in embedded community.

The module has already bluetooth support and I guess it would be beneficial to have also wifi support.

**Comment by ifyall** [comment]
This is in progress in [PR 59444](https://github.com/zephyrproject-rtos/zephyr/pull/59444)! Please review the PR and let us know your thoughts.

**Comment by mariopaja** [comment]
> This is in progress in [PR 59444](https://github.com/zephyrproject-rtos/zephyr/pull/59444)! Please review the PR and let us know your thoughts.

On a brief look, it looks great. However, I noticed that at the moment only SDIO is implemented. Will there be also an implementation for SPI?

**Comment by ifyall** [comment]
> 

@mariopaja, we don't have that in our short/medium term roadmap. We would be open to some collaboration on that!

**Comment by PatrickE94** [comment]
@ifyall I would be open to collaborate on this (adding SPI support). Is anyone else actively working on this?

**Comment by ifyall** [comment]
@PatrickE94, we don't have anyone internally working on it, at least in the Zephyr context. We would love some assistance here!
Ian

**Comment by drensber** [comment]
My company has added PIO-SPI support for the Pico W/RP2040. We may eventually submit code to extend this to the Infineon driver in drivers/wifi/infineon. In the mean time, if someone needs WiFi on the Pico W right away, I have a repo that does an out-of-tree build of my port of the cyw43-driver (the one used in the Pico SDK) here: https://github.com/beechwoods-software/zephyr-cyw43-driver.git
